### PR TITLE
merge from main into dev/4.3.x

### PR DESCRIPTION
**Pull request by LioRoger**
#### What type of PR is this?

type-build
#### Which issue(s) this PR fixes:

Fixes #

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by ungreat — DISMISSED** [review]

**Comment by yhilmare** [comment]
`.gitsubmodules` is wrong, we delete the wrong content

File: .gitmodules [hunk 1]
```diff
@@ -1,7 +1,3 @@
-[submodule "client"]
```
Comment (yizhouxw) [inline .gitmodules:1]: may not delete submodule

**Review by yizhouxw — APPROVED** [review]
submodule recovered now

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**